drivers/SmartThings/matter-sensor/src/init.lua

```diff
@@ -23,6 +23,44 @@ local function device_init(driver, device)
   device:subscribe()
 end
 
+local function do_configure(driver, device)
+  local battery_eps = device:get_endpoints(clusters.PowerSource.ID, {feature_bitmap = clusters.PowerSource.types.PowerSourceFeature.BATTERY})
+  local profile_name = ""
+
+  if device:supports_capability(capabilities.motionSensor) then
+    profile_name = profile_name .. "-motion"
+  end
+
+  if device:supports_capability(capabilities.contactSensor) then
+    profile_name = profile_name .. "-contact"
+  end
+
+  if device:supports_capability(capabilities.illuminanceMeasurement) then
+    profile_name = profile_name .. "-illuminance"
+  end
+
+  if device:supports_capability(capabilities.temperatureMeasurement) then
+    profile_name = profile_name .. "-temperature"
+  end
+
+  if device:supports_capability(capabilities.relativeHumidityMeasurement) then
+    profile_name = profile_name .. "-humidity"
+  end
+
+  if #battery_eps > 0 then
+    profile_name = profile_name .. "-battery"
+  end
+
+  -- remove leading "-"
+  profile_name = string.sub(profile_name, 2)
+
+  device:try_update_metadata({profile = profile_name})
+end
+
+local function info_changed(driver, device, event, args)
+  device:subscribe()
```

Comment:
probably want to restrict this to the profile changing:

`if device.profile.id ~= args.old_st_store.profile.id`

Reply:
Added in the latest change